tests/test_storage.py

```diff
@@ -0,0 +1,201 @@
+from contextlib import nullcontext as does_not_raise
+from io import BytesIO
+import logging
+import pytest
+
+import paramiko
+
+from nightshift.comms.storage import get_credentials, Drive
+from nightshift.ns_exceptions import DriveError
+
+
+def test_get_credentials(mock_sftp_env, sftpserver):
+    assert get_credentials() == (
+        sftpserver.host,
+        "nightshift",
+        "sftp_password",
+        "sierra_dumps_dir",
+        "load_dir",
+        str(sftpserver.port),
+    )
+
+
+@pytest.mark.localtest
+class TestDriveLive:
```

Comment:
Is this class interacting with the "deployed" SFTP server? If it is I'd consider it to be an integration test and move it to a different subdirectory within tests (which is just a preference). The rationale behind that is that in the future you might want to run mocked unit tests in some cases and full integration tests in others (which can make CI/CD pipelines run faster/better if you don't need to run the integration tests all the time)

Reply:
Yes. This `TestDriveLive` interacts with actual/deployed SFTP server (TEST folder on the shared drive in particular). Its current tests are probably not real integration tests (I had too look up what integration tests are ;-P). But it's a good suggestions to clearly mark and separate unit and integration tests. I will create two directories in the tests: 'unit' and 'integration' - hopefully this will to the job.
As I mentioned, I cannot run any Travis tests on this deployed SFTP because of library's network firewall. I don't know how this will play out with CI/CD pipelines in place.